src/methods.ts

```diff
@@ -254,6 +347,11 @@ export class Method extends AbstractMethodField {
    */
   private code: any;
 
+  private numOpcodeExecs = 0;
+  private jitThreshold = 0;
+  private compiledFunctions: Function[] = [];
```

Comment:
Although I don't understand the benefit of caching the function's source, I agree it would be nice to have a JIT specific run function with the `switch` statements. I had tried a JIT specific run function without `switch` and that had not improved any performance.

Will try the "run with switch" idea and push to this PR if it makes a difference.